### PR TITLE
add recipe for openslide

### PR DESCRIPTION
**Pull request by jcupitt**
This is recipe is copy-pasted from:

https://github.com/bioconda/bioconda-recipes/tree/master/recipes/openslide

I'm trying to add a conda package for libvips, and as well as being a general-purpose image processing library, libvips can load images with openslide.

Openslide is currently in bioconda, so I can't list it as a libvips dep in conda-forge. I'm therefore opening this PR as a first step in trying to move openslide from bioconda to conda-forge.

- c/c++ `@conda-forge/help-c-cpp`

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [ ] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/openslide/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    - libtiff >=4
+    - libxml2
+    - openjpeg >=2.1
+    - sqlite >=3.6.2
+    - zlib
+
+test:
+  commands:
+    - test -f $PREFIX/lib/libopenslide.so  # [linux]
+    - test -f $PREFIX/lib/libopenslide.dylib  # [osx]
```
Comment (marcelotrevisani) [inline recipes/openslide/meta.yaml]: You can do it just in one linux, like:
```
- test -f $PREFIX/lib/libopenslide${SHLIB_EXT}  # [unix]
```

File: recipes/openslide/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+about:
+  home: http://openslide.org/
+  license: GNU LGPL 2.1
```
Comment (marcelotrevisani) [inline recipes/openslide/meta.yaml]: I think that should be 
```suggestion
  license: LGPL 2.1
  license_family: LGPL
```

File: recipes/openslide/meta.yaml (hunk 1 above)
Comment (jcupitt) [reply recipes/openslide/meta.yaml]: Nice!

**Comment by jcupitt** [comment]
It's failing to build on win with:

```
... "resolve.py", line 280, in verify_specs
    raise ResolvePackageNotFound(bad_deps)
conda.exceptions.ResolvePackageNotFound: 
  - gdk-pixbuf=2.36.12
```

I'll just tag it as skip for win for now.

File: recipes/openslide/meta.yaml (hunk 1 above)
Comment (marcelotrevisani) [reply recipes/openslide/meta.yaml]: You will need to add some tests for windows, if it is the same kind of test, you can write something like:
```
    - if not exist %LIBRARY_LIB%\libopenslide.lib  exit 1   # [win]
```

File: recipes/openslide/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    - libpng
+    - libtiff >=4
+    - libxml2
+    - openjpeg >=2.1
+    - sqlite >=3.6.2
+    - zlib
+
+test:
+  commands:
+    - test -f $PREFIX/lib/libopenslide${SHLIB_EXT}  # [unix]
```
Comment (marcelotrevisani) [inline recipes/openslide/meta.yaml]:
```suggestion
    - test -f $PREFIX/lib/libopenslide${SHLIB_EXT}  # [unix]
    - if not exist %LIBRARY_LIB%\libopenslide.dll exit 1   # [win]
```

File: recipes/openslide/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
+{% set OPENSLIDE_VERSION = "3.4.1" %}
+
+package:
+  name: openslide
+  version: {{OPENSLIDE_VERSION}}
+
+source:
+  url: https://github.com/openslide/openslide/releases/download/v{{OPENSLIDE_VERSION}}/openslide-{{OPENSLIDE_VERSION}}.tar.gz
+  md5: 88cfe1e9133e206c4e88402b7d19df31
```
Comment (marcelotrevisani) [inline recipes/openslide/meta.yaml]: Could you please change it to use ``sha256`` ?
```suggestion
  sha256: fed08fab8a9b1ded95a34e196652291127ebe392c11f9bc13d26e760295a102d
```

**Comment by marcelotrevisani** [comment]
> It's failing to build on win with:
> 
> ```
> ... "resolve.py", line 280, in verify_specs
>     raise ResolvePackageNotFound(bad_deps)
> conda.exceptions.ResolvePackageNotFound: 
>   - gdk-pixbuf=2.36.12
> ```
> 
> I'll just tag it as skip for win for now.

Sure, no problem! You will need to create this package ``gdk-pixbuf`` for windows

**Comment by jcupitt** [comment]
I'm puzzled by the build failure on win. I don't see how it can have been working on bioconda.

**Comment by marcelotrevisani** [comment]
> I'm puzzled by the build failure on win. I don't see how it can have been working on bioconda.

I think ``bioconda`` does not build packages for windows.

**Comment by jcupitt** [comment]
Ah, that would explain it. Thanks!

If this passes, I'll try removing the xorg deps. I don't think they should be necessary.

**Comment by jcupitt** [comment]
OK, I think this is ready.

I've opened a bioconda PR to remove openslide from there:

https://github.com/bioconda/bioconda-recipes/pull/16938

Once this is merged, that remove could go ahead (if I understand the process).

**Pull request merged**